app/naive_bayes.py

```diff
@@ -0,0 +1,286 @@
+                dict_link_features["Status"] = cur_link.status
+                dict_link_features["Command"] = str(b64decode(cur_link.command).decode('utf-8', errors='ignore').replace('\n', ''))
+                dict_link_features["Number_Facts"] = len(cur_link.used)
+                dict_link_features["Visibility_Score"] = cur_link.visibility.score
+                dict_link_features["Executor_Platform"] = cur_link.executor.platform
+                dict_link_features["Executor_Name"] = cur_link.executor.name
+                
+                agent_paw = cur_link.paw
+                if agent_paw in dict_agent_paw_to_info.keys():
+                    contact_type, trusted_status, privilege, architecture = dict_agent_paw_to_info[agent_paw]
+                    dict_link_features["Agent_Protocol"] = contact_type
+                    dict_link_features["Trusted_Status"] = trusted_status
+                    dict_link_features["Agent_Privilege"] = privilege
+                    dict_link_features["Host_Architecture"] = architecture
+                else:
+                    dict_link_features["Agent_Protocol"] = None
+                    dict_link_features["Trusted_Status"] = None
+                    dict_link_features["Agent_Privilege"] = None
+                    dict_link_features["Host_Architecture"] = None
+
+                dict_link_features["Link_Facts"] = self._get_useful_facts(cur_link)
+
+                link_row = [
+                    dict_link_features["Status"],
+                    dict_link_features["Ability_ID"], 
+                    dict_link_features["Link_Facts"], 
+                    dict_link_features["Planner"],
+                    dict_link_features["Obfuscator"],
+                    dict_link_features["Adversary_ID"],
+                    dict_link_features["Adversary_Name"],
+                    dict_link_features["Command"],
+                    dict_link_features["Number_Facts"],
+                    dict_link_features["Visibility_Score"],
+                    dict_link_features["Executor_Platform"],
+                    dict_link_features["Executor_Name"],
+                    dict_link_features["Agent_Protocol"],
+                    dict_link_features["Trusted_Status"],
+                    dict_link_features["Agent_Privilege"],
+                    dict_link_features["Host_Architecture"]
+                ]
+                matrix_link_data.append(link_row)
+        return matrix_link_data
+
+    def _get_useful_facts(self, cur_link : Link) -> Dict[str, str]:
+        """
+        Selects generalizable facts from link object for data querying
+        """
+        useful_facts = {}
+        if(len(cur_link.used) > 0):
+            for used_fact in cur_link.used:
+                # exclude fact types that are unique to a specific host
+                if str(used_fact.trait).startswith(("host.", "remote.", "file.last.", "domain.user.")) == False:
```

Comment:
You should either make this tuple configurable by the user or define it as a constant near the top of the file in case these things change in the future.